### DISCUSSION TITLE
Schema evolution with read_parquet and hive metadata

### DISCUSSION THREAD

**Discussion by burd0047** [duckdb/duckdb#10673] [Ideas]
I have a case where I want to support schema evolution (new columns) and want to be able to query both newly written and existing parquet files.  Its awesome that one can pass an array of globs to the read_parquet function, however, when using hive metadata, I can't "force" a read/sort order to the files as mentioned in https://github.com/duckdb/duckdb/issues/2877.

I am not sure how to work around this.  Anyone have any ideas? 

Example of my from clause: `read_parquet('./*/*.parquet', hive_partitioning = 1)`
Example of my filename structure: 
`/date=2024-01-01/01.parquet`

I would like to force the read_parquet function to read the newest file first as it would have any new columns.

**Comment by burd0047** [comment]
One thing I have considered is scanning the files myself and doing my own "partition pruning" to generate the list of candidate files to pass to `read_parquet` and sorting those in reverse chronological order.  Any other thoughts?

**Reply by Tishj** [reply]
There's a `glob` scalar function which outputs a list of found filepaths, you could sort/filter those?
But maybe that breaks the hive-partitioning functionality

**Reply by burd0047** [reply]
That could be a workable approach, I could keep the directory name there and still allow hive partitioning to work.  I do wonder about performance impact of a from clause with thousands of files in it vs a glob pattern.
